python/ray/data/impl/pipeline_executor.py

```diff
@@ -29,8 +29,9 @@ def __init__(self, pipeline: "DatasetPipeline[T]"):
         self._stages: List[ObjectRef[Dataset[
             Any]]] = [None] * (len(self._pipeline._stages) + 1)
         self._iter = iter(self._pipeline._base_iterable)
-        self._stages[0] = pipeline_stage.remote(
-            next(self._iter), DatasetContext.get_current())
+        with concurrent.futures.ThreadPoolExecutor() as pool:
+            self._stages[0] = pool.submit(lambda: pipeline_stage(
+                            next(self._iter), DatasetContext.get_current()))
```

Comment:
Btw I don't think that we need to provide the dataset context to the pipeline stage anymore, since `pipeline_stage` are now executing in the same process so the context should already be set.